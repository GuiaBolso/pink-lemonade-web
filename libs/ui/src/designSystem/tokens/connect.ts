import tokens, { TokensGuiaBolsoProps } from './guiaBolso';
import { DeepPartial } from '../../typings/DeepPartial';

export type TokensConnectProps = TokensGuiaBolsoProps &
  DeepPartial<typeof connectTokens>;

const baseFontSize = 16;

const connectTokens = {
  ...tokens,

  name: 'connect',

  type: {
    ...tokens.type,

    baseFontSize,
  },

  colors: {
    ...tokens.colors,

    brand: {
      ...tokens.colors.brand,

      secondary: {
        lightest: '#f2fcfe',
        lighter: '#59d7f2',
        light: '#26caee',
        default: '#00c1eb',
        dark: '#02a6ca',
        darker: '#05829d',
      },
    },
  },
};

export default connectTokens;
