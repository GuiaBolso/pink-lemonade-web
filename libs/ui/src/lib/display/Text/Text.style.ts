import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { CustomThemeProps } from '../../../typings/CustomThemeProps';

type TextProps = {
  variant?: string;
  as?: React.ElementType;
  theme: CustomThemeProps;
};

const variants: object = {
  'display-00': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize80};
        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.light};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'display-01': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize64};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxxl};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'display-02': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize48};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxl};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-01': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize42};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xl};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-02': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize32};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.lg};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-03': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize24};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.md};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-04': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize20};

        color: ${theme?.colors?.neutral.darkest};
        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.sm};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-05': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize18};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xs};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-06': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize14};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-01': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-02': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize14};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-01': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize18};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-02': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'button-text': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  caption: ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize12};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  overline: ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize12};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
        text-transform: uppercase;
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  link: ({ theme }: TextProps) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        color: ${theme?.colors?.neutral?.darkest};
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${`-webkit-${theme?.type?.lineHeight?.rule2}`};
        line-height: ${`-moz-${theme?.type?.lineHeight?.rule2}`};
        line-height: ${theme?.type?.lineHeight?.rule2};
        text-decoration: underline;

        &:hover {
          color: ${theme?.colors?.brand?.secondary?.dark};
        }

        &:visited {
          color: ${theme?.colors?.brand?.primary?.default};
        }
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-medium': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-small': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-medium': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-small': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'overline-medium': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
        text-transform: uppercase;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'overline-small': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
        text-transform: uppercase;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'link-medium': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xs};

        color: ${theme?.colors?.brand?.secondary?.default};
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
        text-decoration: underline;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'link-small': ({ theme }: TextProps) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        color: ${theme?.colors?.brand?.secondary?.default};
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
        text-decoration: underline;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },
};

export const Text = styled.span<TextProps>`
  margin: 0;
  padding: 0;

  /* stylelint-disable-next-line */
  ${props => variants[props.variant]}
`;
