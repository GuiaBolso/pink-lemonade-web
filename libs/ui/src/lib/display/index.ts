export * from './Divider';
export * from './Text';
export * from './Tooltip';
