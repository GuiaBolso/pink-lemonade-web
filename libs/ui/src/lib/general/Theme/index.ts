import { customThemeProps as TCustomThemeProps } from './Theme';

export { default as Theme } from './Theme';
export type customThemeProps = TCustomThemeProps;
