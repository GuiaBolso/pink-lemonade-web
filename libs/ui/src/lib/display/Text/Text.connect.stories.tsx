import React from 'react';
import { Text } from '.';
import { Theme } from '../../general';
import { connect } from '../../../designSystem/tokens';

export default {
  title: 'Display/Text/Connect',
};

export const display00 = () => (
  <Theme theme={connect}>
    <Text variant="display-00">display-00</Text>
  </Theme>
);
export const display01 = () => (
  <Theme theme={connect}>
    <Text variant="display-01">display-01</Text>
  </Theme>
);
export const display02 = () => (
  <Theme theme={connect}>
    <Text variant="display-02">display-02</Text>
  </Theme>
);
export const heading01 = () => (
  <Theme theme={connect}>
    <Text variant="heading-01">heading-01</Text>
  </Theme>
);
export const heading02 = () => (
  <Theme theme={connect}>
    <Text variant="heading-02">heading-02</Text>
  </Theme>
);
export const heading03 = () => (
  <Theme theme={connect}>
    <Text variant="heading-03">heading-03</Text>
  </Theme>
);
export const heading04 = () => (
  <Theme theme={connect}>
    <Text variant="heading-04">heading-04</Text>
  </Theme>
);
export const heading05 = () => (
  <Theme theme={connect}>
    <Text variant="heading-05">heading-05</Text>
  </Theme>
);
export const heading06 = () => (
  <Theme theme={connect}>
    <Text variant="heading-06">heading-06</Text>
  </Theme>
);
export const body01 = () => (
  <Theme theme={connect}>
    <Text variant="body-01">body-01</Text>
  </Theme>
);
export const body02 = () => (
  <Theme theme={connect}>
    <Text variant="body-02">body-02</Text>
  </Theme>
);
export const subtitle01 = () => (
  <Theme theme={connect}>
    <Text variant="subtitle-01">subtitle-01</Text>
  </Theme>
);
export const subtitle02 = () => (
  <Theme theme={connect}>
    <Text variant="subtitle-02">subtitle-02</Text>
  </Theme>
);
export const buttonText = () => (
  <Theme theme={connect}>
    <Text variant="button-text">button-text</Text>
  </Theme>
);
export const caption = () => (
  <Theme theme={connect}>
    <Text variant="caption">caption</Text>
  </Theme>
);
export const overline = () => (
  <Theme theme={connect}>
    <Text variant="overline">overline</Text>
  </Theme>
);
export const link = () => (
  <Theme theme={connect}>
    <Text variant="link" component="a" href="#">
      link
    </Text>
  </Theme>
);
