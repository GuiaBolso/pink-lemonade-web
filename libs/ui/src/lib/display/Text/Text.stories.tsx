import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Text, TextProps } from './Text';

export default {
  title: 'Display/Text',
  component: Text,
} as Meta;

const Template: Story<TextProps> = args => <Text {...args} />;

export const Base = Template.bind({});

Base.args = {
  variant: 'heading-01',
  component: 'p',
  children: 'Text',
};

export const display01 = () => <Text variant="display-01">display-01</Text>;
export const display02 = () => <Text variant="display-02">display-02</Text>;
export const heading01 = () => <Text variant="heading-01">heading-01</Text>;
export const heading02 = () => <Text variant="heading-02">heading-02</Text>;
export const heading03 = () => <Text variant="heading-03">heading-03</Text>;
export const heading04 = () => <Text variant="heading-04">heading-04</Text>;
export const heading05 = () => <Text variant="heading-05">heading-05</Text>;
export const heading06 = () => <Text variant="heading-06">heading-06</Text>;
export const bodyMedium = () => <Text variant="body-medium">body-medium</Text>;
export const bodySmall = () => <Text variant="body-small">body-small</Text>;
export const subtitleMedium = () => (
  <Text variant="subtitle-medium">subtitle-medium</Text>
);
export const subtitleSmall = () => (
  <Text variant="subtitle-small">subtitle-small</Text>
);
export const OverlineMedium = () => (
  <Text variant="overline-medium">overline-medium</Text>
);
export const OverlineSmall = () => (
  <Text variant="overline-small">overline-small</Text>
);
export const LinkMedium = () => (
  <Text variant="link-medium" component="a">
    link-medium
  </Text>
);
export const LinkSmall = () => (
  <Text variant="link-small" component="a">
    link-small
  </Text>
);
