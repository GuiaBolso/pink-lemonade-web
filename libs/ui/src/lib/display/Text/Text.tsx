import React, { memo } from 'react';
import * as Styled from './Text.style';

export type VariantProps =
  | 'display-00'
  | 'display-01'
  | 'display-02'
  | 'heading-01'
  | 'heading-02'
  | 'heading-03'
  | 'heading-04'
  | 'heading-05'
  | 'heading-06'
  | 'body-01'
  | 'body-02'
  | 'subtitle-01'
  | 'subtitle-02'
  | 'button-text'
  | 'caption'
  | 'overline'
  | 'link'
  | 'body-medium'
  | 'body-small'
  | 'subtitle-medium'
  | 'subtitle-small'
  | 'overline-medium'
  | 'overline-small'
  | 'link-medium'
  | 'link-small';

export type TextProps = React.ComponentProps<'span'> & {
  /**
   * Component tag that will be rendered
   */
  component?: React.ElementType; // Component tag that will be rendered (div, p, strong, etc)
  /**
   * Text or Component that will be rendered (string, number, other component)
   */
  children: React.ReactNode;
  /**
   * Variation of style that the component will assume (caption, subtitle-01, link, etc)
   */
  variant?: VariantProps;
};

export const Text = ({
  children,
  component = 'span',
  variant = 'body-01',
  ...rest
}: TextProps) => {
  return (
    <Styled.Text as={component} variant={variant} data-testid="text" {...rest}>
      {children}
    </Styled.Text>
  );
};

export default memo(Text);
