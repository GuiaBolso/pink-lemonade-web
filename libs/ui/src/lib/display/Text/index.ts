import { Text, VariantProps } from './Text';

export { Text, VariantProps };
