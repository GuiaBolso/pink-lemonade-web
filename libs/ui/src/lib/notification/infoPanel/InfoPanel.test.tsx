import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { Theme } from '../../general';
import { connect, guiaBolso } from '../../../designSystem/tokens';

import { InfoPanel } from './index';
import { Text } from '../../display/Text';

describe('InfoPanel Component', () => {
  const InfoPanelComponent = () => (
    <Theme theme={guiaBolso}>
      <InfoPanel>
        <Text
          variant="heading-06"
          component="h3"
          style={{ marginBottom: '10px' }}
        >
          Não precisa autorizar mais nada
        </Text>

        <Text variant="subtitle-small" component="p">
          Talvez você receba um sms do seu banco solicitando para autorizar ou
          habilitar o GuiaBolso. Não precisa e pode ficar tranquilo que não
          faremos nenhum tipo de transação na conta.
        </Text>
      </InfoPanel>
    </Theme>
  );

  const InfoPanelComponentConnect = () => (
    <Theme theme={connect}>
      <InfoPanel>
        <Text
          variant="heading-06"
          component="h3"
          style={{ marginBottom: '10px' }}
        >
          Não precisa autorizar mais nada
        </Text>

        <Text variant="subtitle-small" component="p">
          Talvez você receba um sms do seu banco solicitando para autorizar ou
          habilitar o GuiaBolso. Não precisa e pode ficar tranquilo que não
          faremos nenhum tipo de transação na conta.
        </Text>
      </InfoPanel>
    </Theme>
  );

  it('Renders correctly', () => {
    const tree = renderer.create(<InfoPanelComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders Connect Theme correctly', () => {
    const tree = renderer.create(<InfoPanelComponentConnect />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
