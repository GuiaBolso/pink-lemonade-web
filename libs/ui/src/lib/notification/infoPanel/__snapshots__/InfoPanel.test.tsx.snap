// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel Component Renders Connect Theme correctly 1`] = `
<div
  className="css-aqtonj"
>
  <img
    alt="aviso"
    className="css-totz6w"
    src="test-file-stub"
  />
  <span
    className="css-9wn2e6"
    data-testid="text"
  >
    <h3
      className="css-q33bl2"
      data-testid="text"
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      Não precisa autorizar mais nada
    </h3>
    <p
      className="css-78di6l"
      data-testid="text"
    >
      Talvez você receba um sms do seu banco solicitando para autorizar ou habilitar o GuiaBolso. Não precisa e pode ficar tranquilo que não faremos nenhum tipo de transação na conta.
    </p>
  </span>
</div>
`;

exports[`InfoPanel Component Renders correctly 1`] = `
<div
  className="css-12j4tx"
>
  <img
    alt="aviso"
    className="css-totz6w"
    src="test-file-stub"
  />
  <span
    className="css-1nrdplv"
    data-testid="text"
  >
    <h3
      className="css-1t5ydug"
      data-testid="text"
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      Não precisa autorizar mais nada
    </h3>
    <p
      className="css-1t5ydug"
      data-testid="text"
    >
      Talvez você receba um sms do seu banco solicitando para autorizar ou habilitar o GuiaBolso. Não precisa e pode ficar tranquilo que não faremos nenhum tipo de transação na conta.
    </p>
  </span>
</div>
`;
